### PR TITLE
ocaml-freestanding 0.4.3

### PR DESCRIPTION
**Pull request by mato**

### CODE REVIEW

**Comment by mseri** [comment]
The system switch test fails with
```
#=== ERROR while installing ocaml-freestanding.0.4.3 ==========================#
# context     2.0.0 | linux/x86_64 | ocaml-system.4.06.1 | file:///home/travis/build/ocaml/opam-repository
# path        ~/.opam/ocaml-system/.opam-switch/build/ocaml-freestanding.0.4.3
# command     ~/.opam/opam-init/hooks/sandbox.sh install make install PREFIX=/home/travis/.opam/ocaml-system
# exit-code   2
# env-file    ~/.opam/log/ocaml-freestanding-11425-c743ac.env
# output-file ~/.opam/log/ocaml-freestanding-11425-c743ac.out
### output ###
# [...]
# +cp build/ocaml/byterun/caml/mlvalues.h /home/travis/.opam/ocaml-system/include/ocaml-freestanding/caml/mlvalues.h
# +cp build/ocaml/byterun/caml/printexc.h /home/travis/.opam/ocaml-system/include/ocaml-freestanding/caml/printexc.h
# +cp build/ocaml/byterun/caml/signals.h /home/travis/.opam/ocaml-system/include/ocaml-freestanding/caml/signals.h
# +cp build/ocaml/byterun/caml/compatibility.h /home/travis/.opam/ocaml-system/include/ocaml-freestanding/caml/compatibility.h
# +cp build/ocaml/byterun/caml/bigarray.h /home/travis/.opam/ocaml-system/include/ocaml-freestanding/caml/bigarray.h
# +cp build/ocaml/byterun/caml/m.h /home/travis/.opam/ocaml-system/include/ocaml-freestanding/caml/m.h
# +cp build/ocaml/byterun/caml/s.h /home/travis/.opam/ocaml-system/include/ocaml-freestanding/caml/s.h
# +cp build/ocaml/asmrun/libasmrun.a /home/travis/.opam/ocaml-system/lib/ocaml-freestanding/libasmrun.a
# +ln -s /home/travis/.opam/ocaml-system/lib/ocaml-freestanding/libasmrun.a /home/travis/.opam/ocaml-system/lib/ocaml/libasmrunfreestanding.a
# ln: failed to create symbolic link '/home/travis/.opam/ocaml-system/lib/ocaml/libasmrunfreestanding.a': No such file or directory
# Makefile:122: recipe for target 'install' failed
# make: *** [install] Error 1
```

**Comment by mato** [comment]
@mseri Thanks for digging out the failure. This is due to the changes in mirage/ocaml-freestanding#50, @pascutto / @TheLortex can you look into this please and let me know what you want me to do?

**Comment by TheLortex** [comment]
We had this issue with `mirage-platform` too. The problem is that a symlink was done in the ocaml library path, but that doesn't work in a system switch because of permissions. 
The problem is introduced in https://github.com/mirage/ocaml-freestanding/commit/3dfba6ad0c5b3a8ca1711f8df5910d8528d5bac4 
We found a workaround with the ocaml compiler so you can just revert this commit and the CI should pass.

File: packages/ocaml-freestanding/ocaml-freestanding.0.4.3/opam [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+]
+substs: [
+  "flags/cflags.tmp"
+  "flags/libs.tmp"
+]
+conflicts: [
+  "sexplib" {= "v0.9.0"}
+  "solo5-kernel-ukvm"
+  "solo5-kernel-virtio"
+  "solo5-kernel-muen"
```
Comment (samoht) [inline packages/ocaml-freestanding/ocaml-freestanding.0.4.3/opam]:
```suggestion
  "ocaml-system"
```

Comment (hannesm) [reply packages/ocaml-freestanding/ocaml-freestanding.0.4.3/opam]: I'd rather be in favour to close this PR and release ocaml-freestanding without the symlink (as PR'ed in https://github.com/mirage/ocaml-freestanding/pull/51, same for xen https://github.com/mirage/mirage-platform/pull/208)

Comment (samoht) [reply packages/ocaml-freestanding/ocaml-freestanding.0.4.3/opam]: He someone is keen to do the work, yes sur no pb :p if not we can just merge this PR (as we did for mirage-platform) and wait for the next release to get support for the system compiler.

**Comment by mato** [comment]
I've merged https://github.com/mirage/ocaml-freestanding/pull/51 (thanks @hannesm!) which reverts the symlink changes, will cut an 0.4.4 release shortly and open a new PR for that.

**Pull request closed without merging**